department-of-veterans-affairs/caseflow
# Piedram/appeals 36343

Resolves: [APPEALS-29193](https://jira.devops.va.gov/browse/APPEALS-29193), [APPEALS-29205](https://jira.devops.va.gov/browse/APPEALS-29205), [APPEALS-34996](https://jira.devops.va.gov/browse/APPEALS-34996), [APPEALS-35592](https://jira.devops.va.gov/browse/APPEALS-35592), [APPEALS-35593](https://jira.devops.va.gov/browse/APPEALS-35593), [APPEALS-29194](https://jira.devops.va.gov/browse/APPEALS-29194),
[APPEALS-34916](https://jira.devops.va.gov/browse/APPEALS-34916), [APPEALS-36343](https://jira.devops.va.gov/browse/APPEALS-36343)

# Description
As a Mail Supervisor user I need to be able to reject or accept a Remove Package request so that I am complete the removal process.
 
## Acceptance Criteria

1. The Mail Team Supervisor is able to click a "Review Removal request" button when the package has a pending removal request
2. When a Mail Supervisor clicks on "Review Removal request" button a modal is appears
  2.1. This modal is titled "Review Request" and the below text appears:
    2.1.1. "Reason for Removal:"
   			- The text from the RemovePackageTask will appear
				- "Choose whether to approve the request for removal or reject it."
						- "Approve request" radio button
						- "Reject request" radio button
								A mandatory text box appears titled "Provide a reason for rejection"
								Confirm button is greyed out until text is provided
    2.1.2. The user can click a "Cancel" link
	- This closes the modal and user stays on the Review Package page
    2.1.3. The user can click a "Confirm" button
	- User is directed to the Correspondence Case Queue
	- A banner appears at the top of the queue
	- You have successfully removed a mail package for "veteran name".
	- The package has been removed from Caseflow and must be manually uploaded again from the Centralized Mail Portal, if it needs to be processed. 
	- This button is greyed out until user selected a reason

## Testing Plan

1. Login as a Mail team user `MAIL_TEAM_SUPERVISOR_MAIL_INTAKE_USER (VACO)`
2. From the App selector, select and click on `Queue`
3. In Your cases page, click on `Switch views` and select `Correspondence Cases`
4. Now you have to see the `Correspondence Cases` page
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/507b9ab6-4423-4a21-83b2-d82154d9d41a)

5. Select a package that was previously selected to remove, the review package page has to looks like that this
![image (18)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/beb633ae-1ef3-49ca-ab2e-8aedbcaaf24f)

![image (23)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/de28aa25-8bad-4304-b001-f77615853600)

6. click on `Review removal request` button a modal windows has to show up
![image (19)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/aa32fed9-fc13-4278-9cda-bf1d92b4ddde)

7. In the Modal windows we have two options when you click on the first radio button the button `confirm` is enable
![image (20)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/3d449bc1-4b8b-4ff9-ab1d-90d0bea0070b)

8. In the second option you need to fill a text area up, to enable the button `confirm`
![image (21)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/2506dab8-eeb9-4111-8d35-1486ac4b0076)

9. When clicking on the `Confirm` button the application redirects to the correspondence cases list and shows a banner
![image (22)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/8ca99887-468c-4c32-b303-271818f104ae)

![image (24)](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/bbebdd9f-597c-41cf-aecf-7aa1a31d9e58)
